src/PolyType.SourceGenerator/SourceFormatter/SourceFormatter.Object.cs

```diff
@@ -39,6 +41,25 @@ private void FormatObjectTypeShapeFactory(SourceWriter writer, string methodName
         FormatMemberAccessors(writer, objectShapeModel);
     }
 
+    private static string FormatAssociatedTypeFactory(ObjectShapeModel objectShapeModel)
+    {
+        if (objectShapeModel.AssociatedTypes.Length == 0)
+        {
+            return "null";
+        }
+
+        StringBuilder builder = new();
+        builder.Append("static associatedType => associatedType switch { ");
+        foreach (AssociatedTypeId associatedType in objectShapeModel.AssociatedTypes)
+        {
+            builder.Append($"\"{associatedType.Open}\" or \"{associatedType.Closed}\" => () => new {associatedType.Constructor}(), ");
```

Comment:
Nit: use triple-quote strings to avoid string escaping.

Reply:
I could. But then I'd need to put the content on its own line (especially because the string itself starts with a quote character. And then your the trim trailing whitespace rule would remove the last space on the line, slightly degrading the generated C# code.